src/wasm/wasm-emscripten.cpp

```diff
@@ -577,7 +584,13 @@ struct AsmConstWalker : public LinearExecutionWalker<AsmConstWalker> {
   void queueImport(Name importName, std::string baseSig);
   void addImports();
 
+  template<typename T> Index resolveConstIndex(Expression* arg, T report);
```

Comment:
Instead of templating here, how about `std::function<void(Expression*)>` to make it clearer what report is without looking at the implementation